### PR TITLE
[RFC] scripts/pvscheck.sh: fixes

### PR DESCRIPTION
**Pull request by blueyed**

### CODE REVIEW

File: scripts/pvscheck.sh [hunk 1]
```diff
@@ -1,12 +1,14 @@
-#!/bin/sh
+#!/usr/bin/env bash
```
Comment (jamessan) [inline scripts/pvscheck.sh]: Why did you change this to bash?

Comment (blueyed) [inline scripts/pvscheck.sh]: There were bashisms already, shellcheck suggested it via `In POSIX sh, 'local' is undefined. [SC2039]`.

Comment (jamessan) [inline scripts/pvscheck.sh]: Yes, it's an extension, but it's one that every sh I have access to (dash, posh, mksh, ksh, FreeBSD/OpenBSD/NetBSD sh) implements.  We use that in most of our scripts.

Comment (blueyed) [inline scripts/pvscheck.sh]: Ok.
What about changing it to bash then though anyway?
It's clearer in general, and allows for using `[[` etc.

Comment (justinmk) [inline scripts/pvscheck.sh]: bash isn't available on some bsd systems, so it's good to avoid it unless there is a strong reason not to.

Comment (blueyed) [inline scripts/pvscheck.sh]: I see.
Should we then use the `foo() {(bar=1)}` pattern instead? (used elsewhere already)

Comment (blueyed) [inline scripts/pvscheck.sh]: As for the `local` at hand: if it is really supported everywhere (where relevant), this could be ignored with shellcheck then?!

Comment (justinmk) [inline scripts/pvscheck.sh]: No, just ignore shellcheck here.

Comment (jamessan) [inline scripts/pvscheck.sh]: > if it is really supported everywhere (where relevant), this could be ignored with shellcheck then?!

That's been the assumption so far in our scripts.  If we run into a problem, we can address it then.

File: scripts/pvscheck.sh [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 set -e
 # Note: -u causes problems with posh, it barks at “undefined” $@ when no
 # arguments provided.
 test -z "$POSH_VERSION" && set -u
 
 get_jobs_num() {
-  local num="$(cat /proc/cpuinfo | grep -c "^processor")"
-  num="$(echo $(( num + 1 )))"
+  # TODO: why not only the grep?!
```
Comment (jamessan) [inline scripts/pvscheck.sh]: It's typical to use `numcpus + 1` as the number of parallel jobs.

File: scripts/pvscheck.sh [hunk 3]
```diff
@@ -261,15 +263,6 @@ install_pvs() {(
   rmdir "$pvsdir"
 )}
 
-adjust_path() {
```
Comment (jamessan) [inline scripts/pvscheck.sh]: Instead of removing the function, why not fix it to use `$1` instead of `$tgt`?

Comment (blueyed) [inline scripts/pvscheck.sh]: It does not have to be a function (which you have to look up when reading the source), when it is only used once.

**Review by jamessan — APPROVED** [review]

**Comment by blueyed** [comment]
@jamessan 
Based on my fixup I assume?

**Comment by jamessan** [comment]
> Based on my fixup I assume?

Yup.

**Comment by jamessan** [comment]
There's no need to run ci for this, since it isn't used by anything in this repo.  I'd suggest squashing and adding `[ci skip]`, then merge.

**Comment by blueyed** [comment]
Done.
